### PR TITLE
feat: add mobile device os and browser support

### PR DESCRIPTION
**Pull request by luqven**
This PR adds mobile devices support (iOS and Android) to the library. The compatible devices list was taken from the browser stack [website](https://www.browserstack.com/list-of-browsers-and-platforms/automate).

- adds type definitions for device enums, os, os versions, and browsers
- adds device filtering capabilities
- adds tests to ensure these work as intended
- adds support for macOS
- removes `"jest-package-audit": "^3.1.2"` as a dependency as it's fallen behind on node versions.
- removes the audit tests
## Example

These changes make the following filtering options possible

### Filtering by MacOS version

```js
import { 
  BrowserFilter,
  OperatingSystemFilter,
  OSXOperatingSystemVersionFilter 
}, getCapabilities from 'browserslist-browserstack';

const capabilities = await getCapabilities({
  browsers: {
    include: [BrowserFilter.CHROME],
    exclude: [BrowserFilter.SAFARI, BrowserFilter.IE],
  },
  browserslist: {
    queries: ['ie', 'safari', 'chrome'],
  },
  operatingSystems: {
    include: [OperatingSystemFilter.OSX],
  },
  operatingSystemVersion: {
    include: [OSXOperatingSystemVersionFilter.BIG_SUR],
  }
});
```

### Filtering by iOS device

```js
import { 
  BrowserFilter,
  OperatingSystemFilter,
  OSXOperatingSystemVersionFilter 
}, getCapabilities from 'browserslist-browserstack';

const capabilities = await getCapabilities({
  browsers: {
    include: [BrowserFilter.IPHONE],
  },
  devices: {
    include: [DeviceFilter.IPHONE_XS_MAX],
  }
});
```

### CODE REVIEW

File: source/types.ts [hunk 1]
```diff
@@ -97,4 +126,91 @@ export enum BrowserFilter {
   CHROME = 'chrome',
   OPERA = 'opera',
   EDGE = 'edge',
+  IPHONE = 'iphone',
+  IPAD = 'ipad',
+  ANDROID = 'android',
```
Comment (AndrewLeedham) [inline source/types.ts:131]: Would these not be covered in the OS filter?

**Review by AndrewLeedham — COMMENTED** [review]
This looks great @luqven! Thanks.

With `jest-package-audit`, it really needs some attention, I just don't have the time atm, so removing it for now seems like the best non-blocking approach 👍

File: source/types.ts (hunk 1 above)
Comment (luqven) [reply source/types.ts:131]: So I couldn't make my mind up on this one.

The `os` filter would suffice if I was only interested in `ios` and not a device-specific browser. 

This could look something like

``` js
import { OperatingSystemFilter }, getCapabilities from 'browserslist-browserstack';

const capabilities = await getCapabilities({
  operatingSystems: {
    include: [OperatingSystemFilter.IOS],
  },
});
```

And the resulting bStack API response might look like

```js
  {
      "os": "ios",
      "os_version": "13",
      "browser": "iphone",
      "device": "iPhone XS",
      "browser_version": null,
      "real_mobile": true
  },
...
  {
      "os": "ios",
      "os_version": "13",
      "browser": "ipad",
      "device": "iPad Pro 12.9 2018",
      "browser_version": null,
      "real_mobile": true
  }
```

But what if I only wanted iPhone `ios` browsers?

The OS filter in this case wouldn't be enough.

I could pair it with a `device` filter, but I'd have to list all the iPhone models I want to be tested against rather than test on all supported iPhone models by default.

### OS and Device Filter

``` js
const capabilities = await getCapabilities({
  operatingSystems: {
    include: [OperatingSystemFilter.IOS],
  },
  devices: {
    include: [DeviceFilter.IPHONE_XS_MAX],
  }
});
```
I could try and add an `IPHONE` device filter that includes all iPhone devices. But that would be duplicating what the browser filtering on `iphone` essentially does.

That seems silly if I could instead get all supported iPhone versions with just a `browser` filter.

### Just the browser filter

``` js
import { BrowserFilter }, getCapabilities from 'browserslist-browserstack';

const capabilities = await getCapabilities({
  browsers: {
    include: [BrowserFilter.IPHONE],
  },
});
```

The bStack API response might look like:

```js
  {
      "os": "ios",
      "os_version": "13",
      "browser": "iphone",
      "device": "iPhone XS",
      "browser_version": null,
      "real_mobile": true
  },
...
```

**TLDR**

I'm not sure if we _need_ the browser filter for `iphone`, `ipad` etc. It's a niche use case. But there's enough value there I think to leave it in. Curious to see what you think 😄 .

Comment (AndrewLeedham) [reply source/types.ts:131]: I see. It just reads weird to me since an iPhone isn't a browser. But I guess that is a product of how BrowserStack classifies things. Happy with this 👍

**Comment by luqven** [comment]
> This looks great @luqven! Thanks.
> 
> With `jest-package-audit`, it really needs some attention, I just don't have the time atm, so removing it for now seems like the best non-blocking approach 👍

Thanks, @AndrewLeedham ❤️. I think these tests are failing because the snapshots need to be updated. I can add a commit to that effect or leave that to you all. I Will wait to hear what you think is best.

Once tests pass, is there anything else that needs to happen to get this merged in?

**Comment by AndrewLeedham** [comment]
> Thanks, @AndrewLeedham ❤️. I think these tests are failing because the snapshots need to be updated. I can add a commit to that effect or leave that to you all. I Will wait to hear what you think is best.
> 
> Once tests pass, is there anything else that needs to happen to get this merged in?

Happy for you to update the snapshots. Should be good to merge when tests are passing 👍

**Review by AndrewLeedham — APPROVED** [review]
